### ISSUE TITLE
Changing public link date clunky from desktop

### ISSUE THREAD

**Issue by pjrobertson** [owncloud/client#7058]
### Expected behaviour

Ability to select any date within the range specified by 'default expiration date'

### Actual behaviour

Unable to select dates using the Desktop (MacOS) client v2.5.1

### Steps to reproduce
1. Visit Nextcloud Admin settings: Administration > Sharing > Allow users to share via link > Set default expiration date > Expire after 365 days. **check** 'Enforce Expiration date
2. Right click a synced file on desktop and choose Nextcloud > Sharing...
3. Click 'share via link'
4. The date should be set to exactly 365 days from now (as per settings in step 1)
5. Try and change the date to some time less than 365 days (e.g. 2 months from now).
6. The autocomplete/validation in the field doesn't let you do this.

### Server configuration
Operating system: MacOS 10.14.2
ownCloud version: 15.0.4

**Comment by michaelstingl** [comment]
> * Visit Nextcloud Admin settings…

This is the ownCloud repository. Can you reproduce with ownCloud server?

**Comment by pjrobertson** [comment]
No, the issue is not reproducible on the owncloud server (the server  uses
a calendar date picker to choose the date, which does not have this issue).

You're right that setting must be set on the owncloud server, but the issue
is only reproducible on desktop

**Comment by michaelstingl** [comment]
@pjrobertson I did not ask to test **on** the ownCloud server. It seems your client isn’t connected to a valid ownCloud server. Please test the client **with** an ownCloud server.

**Comment by ogoffart** [comment]
I can reproduce the problem with the owncloud server. It is posdsible the set an arbitrary expiration date

**Comment by ogoffart** [comment]
Actually, it is not possible to set an expiration date which is longer that the enforced one.  So I guess this is working as expected

**Comment by TheOneRing** [comment]
@michaelstingl
